primitives/arithmetic/src/per_things.rs

```diff
@@ -768,6 +769,70 @@ macro_rules! implement_per_thing {
 			}
 		}
 
+		impl Add<Self> for $name {
+			type Output = $name;
+
+			#[inline]
+			fn add(self, rhs: Self) -> Self::Output {
+				$name::from_parts(self.deconstruct().add(rhs.deconstruct()))
```

Comment:
maybe we can do a debug assert for when from_parts is saturating, but I'm fine anyway